src/plugin/sdl/sdl.c

```diff
@@ -398,6 +439,132 @@ static struct bitmap_desc lock_surface(void)
   return BMP(surface->pixels, win_width, win_height, surface->pitch);
 }
 
+#if defined(HAVE_SDL2_TTF) && defined(HAVE_FONTCONFIG)
+
+static int open_font(int psize) {
+  TTF_Font *f;
+
+  assert(sdl_font == NULL);
+
+  SDL_RWseek(sdl_font_rw, 0, RW_SEEK_SET);
+
+  f = TTF_OpenFontRW(sdl_font_rw, sdl_font_idx, psize);
+  if (!f) {
+    error("TTF_OpenFontRW: %s\n", TTF_GetError());
+    return 0;
+  }
+
+  if (!TTF_FontFaceIsFixedWidth(f)) {
+    TTF_CloseFont(f);
+    error("TTF_FontFaceIsFixedWidth: Font is not fixed width\n");
+    return 0;
+  }
+
+  sdl_font = f;
+  sdl_font_size = psize;
+  // get metrics
+  TTF_SizeText(sdl_font, "W", &font_width, NULL);
+  font_height = TTF_FontLineSkip(sdl_font);
+  return 1;
+}
+
+static void close_font(void) {
+  TTF_CloseFont(sdl_font);
+  sdl_font = NULL;
+}
+
+static int setup_ttf_pntsize(int psize) {
+  int ret = 1;
+
+  v_printf("SDL: setup_ttf_pntsize called with size %d\n", psize);
+
+  pthread_mutex_lock(&sdl_font_mtx);
+
+  if (sdl_font)
+    close_font();
+
+  if (!open_font(psize)) {
+    v_printf("SDL: open_font(%d) failed\n", psize);
+    ret = 0;
+  }
+
+  pthread_mutex_unlock(&sdl_font_mtx);
+  return ret;
+}
+
+static int setup_ttf_winsize(int xtarget, int ytarget) {
+  int xnow, ynow;
+  int cols, rows;
+  int i;
+  int ret = 0;
+
+  v_printf("SDL: setup_ttf_winsize called with xtarget %d, ytarget %d\n", xtarget, ytarget);
+
+  // Early on VGA might not be set up, so default;
+  cols = (vga.text_width > 0) ? vga.text_width : 80;
+  rows = (vga.text_height > 0) ? vga.text_height : 25;
```

Comment:
We change window size every time
the video mode is changed.
Before the first video mode set, there
is no window at all.
So if you hit this "too early" then
something is wrong, please call it later.

Reply:
So far I have only found it too early when creating the initial SDL window in `SDL_text_init()`, called from `SDL_init()`. If we have specified window size x/y then I need it to calculate the initial font size, if we have no x/y I need it to calculate x/y from a default font size.

Reply:
> If we have specified window size x/y then I need it to calculate the initial font size,
> if we have no x/y I need it to calculate x/y from a default font size.

Then you can just move that to SDL_change_mode().
There is already the "late initialization" part under
`if (!initialized) {`